pkgs/applications/video/kodi/addons/upnext/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, rel, buildKodiAddon, fetchzip, addonUpdateScript }:
+
+buildKodiAddon rec {
+  pname = "upnext";
+  namespace = "service.upnext";
+  version = "1.1.9+matrix.1";
+
+  src = fetchzip {
+    url = "https://mirrors.kodi.tv/addons/${lib.toLower rel}/${namespace}/${namespace}-${version}.zip";
```

Comment:
can you explain easier to maintain? pulling from `kodi` repo guarantees this should have no quirky build issues

for addons which actually require a "build" step your comment seems relevant but in this case i don't think there is an issue here

Reply:
My idea was that setting both the source Url and the homepage to the same Github project introduces the "single source of truth" for both automated scripts and humans which is easily verifiable.

I did, however, confuse (sorry!) `addonUpdateScript` with the one pulling information from Github, so I thought neither the script nor humans could understand where this package is coming from, when was it released and what are the changes.
This information is not easily obtainable from the wiki-page.

So, yes **this url is ok**.

"homepage" still needs to be set to Github to provide an easy clue for human maintainers.

I want to still point out, though, that this is _fragile_ because we don't actually know where Kodi is pulling the sources for the package (or do we? I haven't found it). One day it might be some other place, not the homepage we set here.
This is a rather philosophical question of what we're _really_ packaging here: some "blob" that Kodi provides under the name "upnext", or some very specific project from Github:-)

I get your point about pulling from Kodi's repo – this seems to be overall the better solution for now, my proposal will lack something else.

Reply:
> "homepage" still needs to be set to Github to provide an easy clue for human maintainers.

i agree, good suggestion 👍



> we don't actually know where Kodi is pulling the sources for the package (or do we? I haven't found it).

they advertise this in the large xml file our updater uses... which might mean we could automatically update the value? i forget... but worth looking into perhaps 🤔



> some "blob" that Kodi provides under the name "upnext"

absolutely!  the `kodi` addon framework is truly a hot mess from a technical point of view and even moreso if you're a nix person - for example most skins modify their own source code as they run which is why we can't package them 🫠

i wish we could do better but when the system is as it is... i think we should accept how it is and try to provide the best value we can on this space

---

thanks for your comments @nvmd!

Reply:
FYI, the kodi addons blobs found in the repository are the one submitted to the github repos you can find here:  https://kodi.wiki/view/Official_add-on_repository

So the "problem" here to me is that the addon developers didn't submit the needed PRs to kodi, not the other way around.

Reply:
> FYI, the kodi addons blobs found in the repository are the one submitted to the github repos you can find here: https://kodi.wiki/view/Official_add-on_repository
> 
> So the "problem" here to me is that the addon developers didn't submit the needed PRs to kodi, not the other way around.

Thanks for the link!
That's interesting. They're just copying the addon files to their own repo...